unit-tests.gs

```diff
@@ -81,3 +81,29 @@ function testSemVer () {
     assert(v2.compare(v1) === (-comp.result), 'Comparison between ' + comp.v2 + ' and ' + comp.v1 + ' did not return the expected value of ' + (-comp.result));
   });
 }
+
+/**
+ * Test all events from selected persiod. It won't send actual mails to you, but put content of them into log.
+ *
+ * !!! Execution of this method very ofted exceed maximum time.
+ *
+ * @param testDate - First date to start test; by default 01.01.2017.
+ * @param numberOfDaysToTest - Number of days to preform test; by default 1 year.
+ */
+function testSelectedPeriod (testDate, numberOfDaysToTest) {
+
+  testDate = testDate || new Date('2017/01/01 06:00:00');
+  numberOfDaysToTest = numberOfDaysToTest || 365;
+
+  log.add('testWholeYear() running.', 'info');
+
+  for (var i = 0; i < numberOfDaysToTest; i++) {
+    var mailContent = generateEmailNotification(testDate);
+    log.add('Subject: ' + mailContent.subject);
+    log.add('Content: ' + mailContent.body);
+    //log.add('HTML Content: ' + mailContent.htmlBody); // Uncomment for printing HTML content into log as well.
```

Comment:
Excluding HTML output by default is OK to me, because manually checking multiple HTML emails is nearly impossible and the probability of a HTML-but-not-plaintext error are really slim: I don't know if adding another parameter to the function to eventually enable the HTML output would be worth the work, so I'll leave the decision up to you: do what you consider the best.

Reply:
I've done it as another parameter.